pydantic_ai_slim/pydantic_ai/agent/__init__.py

```diff
@@ -1333,6 +1344,37 @@ def _normalize_instructions(
             return [instructions]
         return list(instructions)
 
+    def _apply_force_first_request(self, model_settings: ModelSettings | None) -> ModelSettings | None:
+        """Apply force_first_request tool flags to model_settings.
+
+        If any tools have force_first_request=True and no callable tool_choice is set,
+        generate a callable that forces those tools on the first request.
+        """
+        force_first_tools = [tool.name for tool in self._function_toolset.tools.values() if tool.force_first_request]
```

Comment:
One of the reasons why I don't like `force_first_request` on the tool is that it won't work for tools in other toolsets, or e.g. MCP tools